### PR TITLE
[Clang][ASTMatcher] Add `dependentScopeDeclRefExpr` Matcher

### LINKED ISSUE #120937: Missing AST matcher: dependentScopeDeclRefExpr
The AST matchers library seems to be missing a matcher that matches an expression of type `DependentScopeDeclRefExpr`.

### PR DESCRIPTION
**Pull request by AmrDeveloper**
Add AST Matcher for `DependentScopeDeclRefExpr`

Fixes: #120937

### CODE REVIEW

File: clang/unittests/ASTMatchers/ASTMatchersNodeTest.cpp [hunk 1]
```diff
@@ -556,6 +556,22 @@ TEST_P(ASTMatchersTest, DeclRefExpr) {
                          Reference));
 }
 
+TEST_P(ASTMatchersTest, DependentScopeDeclRefExpr) {
+  if (!GetParam().isCXX() || GetParam().hasDelayedTemplateParsing()) {
+    // FIXME: Add a test for `dependentScopeDeclRefExpr()` that does not depend
```
Comment (HighCommander4) [inline clang/unittests/ASTMatchers/ASTMatchersNodeTest.cpp:561]: The concept of template-dependence (and thus `DependentScopeDeclRefExpr` nodes) only exist in C++, so it's expected that the test depends on C++.

Based on some other tests that check this condition, please revise the comment to instead say:

```
// FIXME: Fix this test to work with delayed template parsing.
```

**Review by HighCommander4 — COMMENTED** [review]
Thanks, this looks pretty good!

File: clang/unittests/ASTMatchers/ASTMatchersNodeTest.cpp (hunk 1 above)
Comment (AmrDeveloper) [reply clang/unittests/ASTMatchers/ASTMatchersNodeTest.cpp]: Thank you, I updated it :D

**Review by HighCommander4 — APPROVED** [review]
Thanks, LGTM!

I will wait a few days before merging in case anyone else has feedback.

**Comment by HighCommander4** [comment]
Went ahead and merged this. Thanks @AmrDeveloper!

**Comment by HighCommander4** [comment]
By the way, in case you're interested in doing another one, I filed https://github.com/llvm/llvm-project/issues/121240 about another missing matcher I came across.

**Comment by AmrDeveloper** [comment]
> #121240

Sure, i am interested, Thank you :D

**Pull request merged**